cmd/k8s-operator/proxygroup_specs.go

```diff
@@ -12,169 +12,159 @@ import (
-										Name: "POD_IP",
-										ValueFrom: &corev1.EnvVarSource{
-											FieldRef: &corev1.ObjectFieldSelector{
-												FieldPath: "status.podIP",
-											},
-										},
-									},
-									{
-										Name: "POD_NAME",
-										ValueFrom: &corev1.EnvVarSource{
-											FieldRef: &corev1.ObjectFieldSelector{
-												// Secret is named after the pod.
-												FieldPath: "metadata.name",
-											},
-										},
-									},
-									{
-										Name:  "TS_KUBE_SECRET",
-										Value: "$(POD_NAME)",
-									},
-									{
-										Name:  "TS_STATE",
-										Value: "kube:$(POD_NAME)",
-									},
-									{
-										Name:  "TS_EXPERIMENTAL_VERSIONED_CONFIG_DIR",
-										Value: "/etc/tsconfig/$(POD_NAME)",
-									},
-									{
-										Name:  "TS_USERSPACE",
-										Value: "false",
-									},
-								}
-								if pg.Spec.Type == tsapi.ProxyGroupTypeEgress {
-									envs = append(envs, corev1.EnvVar{
-										Name:  "TS_EGRESS_SERVICES_CONFIG_PATH",
-										Value: fmt.Sprintf("/etc/proxies/%s", egressservices.KeyEgressServices),
-									})
-								}
+		return volumes
+	}()
 
-								if tsFirewallMode != "" {
-									envs = append(envs, corev1.EnvVar{
-										Name:  "TS_DEBUG_FIREWALL_MODE",
-										Value: tsFirewallMode,
-									})
-								}
+	// Main container config.
+	c := &ss.Spec.Template.Spec.Containers[0]
+	c.Image = image
+	c.VolumeMounts = func() []corev1.VolumeMount {
+		var mounts []corev1.VolumeMount
+		for i := range pgReplicas(pg) {
+			mounts = append(mounts, corev1.VolumeMount{
+				Name:      fmt.Sprintf("tailscaledconfig-%d", i),
+				ReadOnly:  true,
+				MountPath: fmt.Sprintf("/etc/tsconfig/%s-%d", pg.Name, i),
+			})
+		}
 
-								return envs
-							}(),
-						},
-					},
-					Volumes: func() []corev1.Volume {
-						var volumes []corev1.Volume
-						for i := range pgReplicas(pg) {
-							volumes = append(volumes, corev1.Volume{
-								Name: fmt.Sprintf("tailscaledconfig-%d", i),
-								VolumeSource: corev1.VolumeSource{
-									Secret: &corev1.SecretVolumeSource{
-										SecretName: fmt.Sprintf("%s-%d-config", pg.Name, i),
-									},
-								},
-							})
-						}
-						if pg.Spec.Type == tsapi.ProxyGroupTypeEgress {
-							volumes = append(volumes, corev1.Volume{
-								Name: pgEgressCMName(pg.Name),
-								VolumeSource: corev1.VolumeSource{
-									ConfigMap: &corev1.ConfigMapVolumeSource{
-										LocalObjectReference: corev1.LocalObjectReference{
-											Name: pgEgressCMName(pg.Name),
-										},
-									},
-								},
-							})
-						}
+		if pg.Spec.Type == tsapi.ProxyGroupTypeEgress {
+			mounts = append(mounts, corev1.VolumeMount{
+				Name:      pgEgressCMName(pg.Name),
+				MountPath: "/etc/proxies",
+				ReadOnly:  true,
+			})
+		}
 
-						return volumes
-					}(),
+		return mounts
+	}()
+	c.Env = func() []corev1.EnvVar {
+		envs := []corev1.EnvVar{
+			{
+				Name: "POD_IP",
```

Comment:
Sorry, since I merged the PR that now reads IPs from an `POD_IPS` set from `.status.podIPs`, we'll need to change it here, else it will overwrite that change